### ISSUE TITLE
`missing-override-decorator` suggests `@typing.override` at `python-version` < 3.12, where `typing.override` does not exist

### ISSUE THREAD

**Issue by AlekseiSokolov2TR** [astral-sh/ty#4085]
### Summary

`missing-override-decorator` fires at `python-version = 3.11` and its `info` sub-diagnostic tells you to *"Decorate the method with `@typing.override`"*. But `typing.override` was added in Python 3.12 (PEP 698). At `python-version < 3.12`, following that suggestion is itself an error — ty flags `from typing import override` as `unresolved-import`. So the diagnostic recommends a fix that the same run reports as invalid for the configured version.

The rule's fix suggestion does not respect the configured `python-version`. Below 3.12 it should point to `typing_extensions.override` (which ty *does* accept — see case 3 below), or the wording should otherwise account for the target version.

**Minimal reproduction**

`base.py`:
```python
class Base:
    def dispatch(self) -> int:
        return 1


class Child(Base):
    def dispatch(self) -> int:
        return 2
```

Command:
```
ty check --error all --python-version 3.11 base.py
```

Output:
```
error[missing-override-decorator]: Method `dispatch` overrides `Base.dispatch` but is not decorated with `@override`
 --> base.py:7:9
  |
7 |     def dispatch(self) -> int:
  |         ^^^^^^^^
info: Decorate the method with `@typing.override` to make the override explicit
```

**The suggested fix is rejected at the same version.** Adding `from typing import override` + `@override`:
```
error[unresolved-import]: Module `typing` has no member `override`
info: The member may be available on other Python versions or platforms
info: Python 3.11 was assumed when resolving imports because it was specified on the command line

error[missing-override-decorator]: ... (still fires)
```

**Only `typing_extensions.override` satisfies the rule at 3.11.** Using `from typing_extensions import override` + `@override` → `All checks passed!`

### Expected

At `python-version < 3.12`, the `missing-override-decorator` info message should suggest `@typing_extensions.override` (the only form available before 3.12), rather than `@typing.override` — which the same run reports as `unresolved-import`.

### Version

`ty 0.0.63`

**Comment by carljm** [comment]
Thanks! Attached PR fixes this.
